tests/test_hubspot_newfw_all_fields.py

```diff
@@ -0,0 +1,112 @@
+
+        for stream, records in self.expected_records.items():
+            LOGGER.info("The test client found %s %s records.", len(records), stream)
+
+        super().setUp()
+        self.convert_datatype(self.expected_records)
+
+    def convert_datatype(self, expected_records):
+        # Convert the time stamp data type, Get keys with data and with no data 
+        for stream, records in expected_records.items():
+            expected_keys = set()
+            for record in records:
+
+                expected_keys = expected_keys.union(record.keys())
+                # convert timestamps to string formatted datetime
+                timestamp_keys = {'timestamp'}
+                for key in timestamp_keys:
+                    timestamp = record.get(key)
+                    if timestamp:
+                        record[key]=self.datetime_from_timestamp(timestamp/1000, str_format=self.BASIC_DATE_FORMAT)
+
+            self.KEYS_WITH_NO_DATA[stream] = self.selected_fields.get(stream).difference(expected_keys)
+
+        return expected_records
+
+    def remove_bad_keys(self, stream):
+        # NB: The following woraround is for dynamic fields on the `deals` stream that we just can't track.
+        #     At the time of implementation there is no customer feedback indicating that these dynamic fields
+        #     would prove useful to an end user. The ones that we replicated with the test client are specific
+        #     to our test data. We have determined that the filtering of these fields is an expected behavior.
+        # deals workaround for 'property_hs_date_entered_<property>' fields
+        # BUG_TDL-14993 | https://jira.talendforge.org/browse/TDL-14993
+        #                 Has an value of object with key 'value' and value 'Null'
+        if stream == 'deals':
+            bad_key_prefixes = {'property_hs_date_entered_', 'property_hs_date_exited_', 'property_hs_time_in'}
+            bad_keys = set()
+            for key in self.expected_all_keys:
+                for bad_prefix in bad_key_prefixes:
+                   if key.startswith(bad_prefix) and key not in self.fields_replicated:
+                        bad_keys.add(key)
+            for key in self.fields_replicated:
+                for bad_prefix in bad_key_prefixes:
+                   if key.startswith(bad_prefix) and key not in self.expected_all_keys:
+                        bad_keys.add(key)
+
+            for key in bad_keys:
+                if key in self.expected_all_keys:
+                    self.expected_all_keys.remove(key)
+                elif key in self.fields_replicated:
+                    self.fields_replicated.remove(key)
+
+    ##########################################################################
+    # Tests To Skip
+    ##########################################################################
+
+    @unittest.skip("Skip till all cards of missing fields are fixed. TDL-16145 ")
```

Comment:
This would be nice place to use the tap-tester @skipUntilDone to ensure this gets picked up as soon as TDL-16145 is completed.